File: src/main/kotlin/dev/jefy/connectpro/portfolio/applicaion/dtos/ProfessionalInfoResponse.java
package dev.jefy.connectpro.portfolio.applicaion.dtos;

import java.util.List;

import dev.jefy.connectpro.portfolio.domain.vo.Availability;
import dev.jefy.connectpro.portfolio.domain.vo.ProfessionalInfo;

/**
 * @author Jôph Yamba
 */
public record ProfessionalInfoResponse(
    int activeYears,
    int numberOfEmployees,
    List<ProjectResponse> projects,
    List<Availability> availabilities
) {
    public static ProfessionalInfoResponse from(ProfessionalInfo professionalInfo, List<ProjectResponse> projects) {
        return new ProfessionalInfoResponse(
                professionalInfo.activeYears(),
                professionalInfo.numberOfEmployees(),
                projects,
                professionalInfo.availabilities()
        );
    }
}

File: src/main/kotlin/dev/jefy/connectpro/portfolio/application/dtos/ProfessionalInfoResponse.kt
package dev.jefy.connectpro.portfolio.application.dtos

import dev.jefy.connectpro.portfolio.domain.vo.Availability
import dev.jefy.connectpro.portfolio.domain.vo.ProfessionalInfo

/**
 * @author Jôph Yamba
 */
@JvmRecord
data class ProfessionalInfoResponse(
    val activeYears: Int?,
    val numberOfEmployees: Int?,
    val projects: List<ProjectResponse>,
    val availabilities: List<Availability>
)

fun ProfessionalInfo.toResponse(projects: List<ProjectResponse>) =  ProfessionalInfoResponse(
        activeYears = this.activeYears,
        numberOfEmployees = this.numberOfEmployees,
        projects = projects,
        availabilities = this.availabilities
)